src/Approximators/Selectors/qrcp.jl

```diff
@@ -0,0 +1,94 @@
+"""
+    QRCP <: Selector
+
+A `Selector` that implements QR with column norm pivoting for selecting column indices from  
+a matrix.
+
+# Fields
+- `compressor::Compressor`, the compression technique that will applied to the matrix, 
+    before selecting indices.
+
+# Constructor
+    QRCP(;compressor = Identity())
+
+## Keywords
+- `compressor::Compressor`, the compression technique that will applied to the matrix, 
+    before selecting indices. Defaults the `Identity` compressor.
+
+## Returns
+- Will return a `QRCP` object.
+"""
+mutable struct QRCP <: Selector
+    compressor::Compressor
+end
+
+function QRCP(;compressor = Identity()) 
+    QRCP(compressor)
+end
+
+"""
+    QRCPRecipe <: SelectorRecipe
+
+A `SelectorRecipe` that contains all the necessary preallocations for selecting column 
+indices from a matrix using QR with column norm pivoting.
+
+# Fields
+- `compressor::Compressor`, the compression technique that will applied to the matrix, 
+    before selecting indices.
+- `SA::AbstractMatrix`, a buffer matrix for storing the sketched matrix.
+"""
+mutable struct QRCPRecipe <: SelectorRecipe
+    compressor::CompressorRecipe
+    SA::AbstractMatrix
+end
+
+function complete_selector(ingredients::QRCP, A::AbstractMatrix)
+    compressor = complete_compressor(ingredients.compressor, A)
+    n_rows, n_cols = size(compressor)
+    SA = Matrix{eltype(A)}(undef, n_rows, n_cols)
```

Comment:
I need some clarification on the dimensions here. If `SA` is the compressed matrix,
which represents `compressor * A` how comes it dimensions are not `size(compressor, 1)` by 
`size(A, 2)`?